projects/knora-ui/src/lib/viewer/values/time-value/time-value.component.ts

```diff
@@ -0,0 +1,198 @@
+    if(this.valueFormControl.value.date instanceof KnoraDate){
+      this.updateDate = new Date(this.valueFormControl.value.date.year,
+                                (this.valueFormControl.value.date.month - 1),
+                                this.valueFormControl.value.date.day,
+                                splitTime[0],
+                                splitTime[1]
+      );
+    } else { // when the user submits a new value, it will be an instance of GregorianCalendarDate
+      this.updateDate = new Date(this.valueFormControl.value.date.calendarStart.year,
+                                (this.valueFormControl.value.date.calendarStart.month - 1),
+                                this.valueFormControl.value.date.calendarStart.day,
+                                splitTime[0],
+                                splitTime[1]
+      );
+    }
+
+    // convert the updateDate to a timestamp so that knora will accept it
+    newTimeValue.time = this.updateDate.toISOString();
+
+    if (this.commentFormControl.value !== null && this.commentFormControl.value !== '') {
+      newTimeValue.valueHasComment = this.commentFormControl.value;
+    }
+
+    return newTimeValue;
+  }
+
+  getUpdatedValue(): UpdateTimeValue | false {
+    if (this.mode !== 'update' || !this.form.valid) {
+      return false;
+    }
+
+    const updatedTimeValue = new UpdateTimeValue();
+
+    updatedTimeValue.id = this.displayValue.id;
+
+    // split the time entry in two to separate the hours and the minutes
+    let splitTime = this.valueFormControl.value.time.split(":");
+
+    // when the first value is displayed on page load, it will be an instance of KnoraDate
+    if(this.valueFormControl.value.date instanceof KnoraDate){
+      this.updateDate = new Date(this.valueFormControl.value.date.year,
+                                (this.valueFormControl.value.date.month - 1),
+                                this.valueFormControl.value.date.day,
+                                splitTime[0],
+                                splitTime[1]
+      );
+    } else { // when the user submits a new value, it will be an instance of GregorianCalendarDate
+      this.updateDate = new Date(this.valueFormControl.value.date.calendarStart.year,
+                                (this.valueFormControl.value.date.calendarStart.month - 1),
+                                this.valueFormControl.value.date.calendarStart.day,
+                                splitTime[0],
+                                splitTime[1]
+      );
+    }
+
+    // convert the updateDate to a timestamp so that knora will accept it
+    updatedTimeValue.time = this.updateDate.toISOString();
+
+    // add the submitted comment to updatedTimeValue only if user has added a comment
+    if (this.commentFormControl.value !== null && this.commentFormControl.value !== '') {
+      updatedTimeValue.valueHasComment = this.commentFormControl.value;
+    }
+
+    return updatedTimeValue;
+  }
+
+  triggerResize() {
```

Comment:
Is this used?

Reply:
It is used for the textarea to automatically resize but as we've agreed I will move all the CSS and HTML changes to a new PR